staging/src/kubevirt.io/api/core/v1/defaults.go

```diff
@@ -75,8 +77,13 @@ func SetDefaults_DiskDevice(obj *DiskDevice) {
 }
 
 func SetDefaults_Watchdog(obj *Watchdog) {
-	if obj.I6300ESB == nil {
-		obj.I6300ESB = &I6300ESBWatchdog{}
+	// Ensure both watchdogs are nil before setting a default
+	if obj.I6300ESB == nil && obj.Diag288 == nil {
+		if runtime.GOARCH == "s390x" {
```

Comment:
@HarshithaMS005 @jschintag are we sure that this is correct? 
I would expect this to rely on something like `vmi.Spec.Architecture`.

Reply:
Made necessary code changes added code to architecture specific defaults files